### PR TITLE
refactor transform partition values

### PR DESCRIPTION
**Pull request by zzeekk**
see #264, improvements are:
- refactoring of transformation of partition values to support: 
  - use Seq[Map[PartitionValues,PartitionValues]] as return type of CustomDfTransformer.transformPartitionValues to make define rename & aggregation of partition values precisely,
  - keep input & outputSubFeed seperated in SparkAction
- implement PartitionDiffMode selectAdditionalInputExpression(#264)
- implement PartitionDiffMode stopIfNoData

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/definitions/ExecutionMode.scala [hunk 1]
```diff
@@ -105,16 +107,28 @@ private[smartdatalake] trait ExecutionModeWithMainInputOutput {
...
- *                                  Default is to return the originally selected partitions found in attribute selectedPartitions.
+ * @param stopIfNoData optional setting if further actions should be skipped if this action has no data to process (default).
+ *                     Set stopIfNoData=false if you want to run further actions nevertheless. They will receive output dataObject unfiltered as input.
+ * @param selectExpression          optional expression to define or refine the list of selected output partitions. Define a spark sql expression working with the attributes of [[PartitionDiffModeExpressionData]] returning a list<map<string,string>>.
+ *                                  Default is to return the originally selected output partitions found in attribute selectedPartitionValues.
+ * @param applyPartitionValuesTransform If true applies the partition values transform of custom transformations on input partition values before comparision with output partition values.
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/definitions/ExecutionMode.scala:114]:
```suggestion
 * @param applyPartitionValuesTransform If true applies the partition values transform of custom transformations on input partition values before comparison with output partition values.
```

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileSubFeedAction.scala [hunk 2]
```diff
@@ -44,102 +44,85 @@ abstract class FileSubFeedAction extends Action {
   override def recursiveInputs: Seq[FileRefDataObject with CanCreateInputStream] = Seq()
 
   /**
-   * Initialize Action with a given [[FileSubFeed]]
-   * Note that this only checks the prerequisits to do the processing and simulates the output FileRef's that would be created.
+   * "Transformes" a given [[FileSubFeed]]
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileSubFeedAction.scala:47]:
```suggestion
   * "Transforms" a given [[FileSubFeed]]
```

File: sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala [hunk 3]
```diff
@@ -176,6 +202,10 @@ case class JdbcTableDataObject(override val id: DataObjectId,
   override def isDbExisting(implicit session: SparkSession): Boolean = connection.catalog.isDbExisting(table.db.get)
   override def isTableExisting(implicit session: SparkSession): Boolean = connection.catalog.isTableExisting(table.db.get, table.name)
 
+  def deleteTable(implicit session: SparkSession): Unit = {
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala:205]: This name is a bit misleading: Sounds like deleting the table itself where in fact you only delete FROM table.

**Review by pgruetter — APPROVED** [review]
Quite the rewrite! Just small comments, but please check the method rename in JdbcTableDataObject before merging.

File: sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala (hunk 3 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala:205]: Ok, fixed to deleteAllData

**Comment by zzeekk** [comment]
Also merged to develop

**Pull request merged**